tests/schema/tests.py

```diff
@@ -532,6 +532,18 @@ def test_alter_text_field(self):
         with connection.schema_editor() as editor:
             editor.alter_field(Note, old_field, new_field, strict=True)
 
+    def test_alter_fk_checks_deferred_constraints(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Node)
+        old_field = Node._meta.get_field('parent')
+        new_field = ForeignKey(Node, CASCADE)
+        new_field.set_attributes_from_name('parent')
+        parent = Node.objects.create()
+        with atomic(), connection.schema_editor() as editor:
+            # Update the parent FK to create a deferred constraint check.
+            Node.objects.update(parent=parent)
+            editor.alter_field(Node, old_field, new_field, strict=True)
```

Comment:
Maybe it's worth an additional comment above the last line to the effect of: `# If constraints aren't set to IMMEDIATE, this operation will fail.` (if that's correct)